### PR TITLE
Browser compatibility

### LINKED ISSUE #20: Browser compatibility?

### PR DESCRIPTION
**Pull request by noveens**
Fixes #20 

Introduced a make end point which will automatically make a browser compatible JS dist under the browser folder by the name, ```bundle.js```.

### CODE REVIEW

**Comment by noveens** [comment]
@PVince81 , 
Is this okay?

File: Makefile [hunk 1]
```diff
@@ -39,9 +39,24 @@ jsdocs:
 	#Output the final documentation link
 	echo "To read the documentation, click here : file://"${DIR}"/jsdoc/ownCloud.html"
 
+browser:
+	sed -i "s/require('fs/require('fs-web/g" owncloud/*.js
```
Comment (PVince81) [inline Makefile]: `sed` sounds a bit barbaric here. Is there any best practice here that could be used ?

**Comment by PVince81** [comment]
I suggest to not check in any generated files. Add them to `.gitignore`.

Let me people generate the bundles with make if they use the git version.
Later on you'll provide these precompiled bundles as release tarballs.

File: owncloud/helperFunctions.js [hunk 2]
```diff
@@ -768,4 +774,26 @@ helpers.prototype._getFileName = function(path) {
...
 
+helpers.prototype._cleanseJson = function(json) {
+	for (var key in json) {
+		var a = recursiveCleanse(json[key]);
+		json[key] = a;
+	}
+	return json;
+};
+
+function recursiveCleanse(json) {
```
Comment (PVince81) [inline owncloud/helperFunctions.js]: what does this do any why ?

File: owncloud/owncloud.js [hunk 3]
```diff
@@ -76,6 +83,34 @@ ownCloud.prototype.login = function(username, password) {
 };
 
 /**
+ * Logs in to the specified ownCloud instance (Updates capabilities)
+ * @param 		{string} 			  instance 		URL of the OC instance
```
Comment (PVince81) [inline owncloud/owncloud.js]: weird indents here

**Comment by PVince81** [comment]
otherwise this looks fine, I didn't test it 👍

File: owncloud/helperFunctions.js (hunk 2 above)
Comment (noveens) [inline owncloud/helperFunctions.js]: That's just for parsing the XML body to a JSON object.

File: Makefile (hunk 1 above)
Comment (noveens) [inline Makefile]: Why so?
Plus, I found many other solutions for the same, but couldn't find any for webPack.

**Pull request merged**